mmdata.c

```diff
@@ -2683,15 +2664,14 @@ long pntrRevInstr(long start_position,pntrString *string1,
     pntrString *string2)
 {
    long ls1,ls2;
-   pntrString *tmp = NULL_PNTRSTRING;
    ls1=pntrLen(string1);
    ls2=pntrLen(string2);
    if (start_position>ls1-ls2+1) start_position=ls1-ls2+2;
    if (start_position<1) return 0;
    while (!pntrEq(string2,pntrMid(string1,start_position,ls2))) {
      start_position--;
-     pntrLet(&tmp,NULL_PNTRSTRING);
-                /* Clear pntrString buffer to prevent overflow caused by "mid" */
+     pntrTempAlloc(0);
```

Comment:
Again: change this in a different PR

Reply:
Addressed on PR comment

Reply:
See my previous remark.  As a bare minimum you must formulate a precondition no caller has allocated something on the stack that it wants to reuse after the call to this function.

Reply:
Every call to `let` has this same effect, so if the code is broken now it was already broken before. Except in very specialized circumstances (specifically, inside the implementation of some of the temp-producing functions), the temp stack is always completely cleared after every `let`, so the lifetime of temporaries in most of the code is syntactically evident (from the temp-producing call to the next call to `let` or `freeTempAlloc`).

Reply:
This is documented here:

https://github.com/metamath/metamath-exe/blob/8b442fa82600ea7892417fe0da17acd12fc3fbf8/src/mmvstr.c#L41-L43

Reply:
Here's another example that shows that these sections are handled with care:

https://github.com/metamath/metamath-exe/blob/8b442fa82600ea7892417fe0da17acd12fc3fbf8/src/mminou.c#L386-L398